contracts/rosca.sol

```diff
@@ -1,72 +1,266 @@
+    uint startTime;
+    int16 feeToService;
+    address feeAddress;
+    uint currentRound;  //currentRound will be set to 0 when ROSCA is created and will turn to one when the ROSCA actually start
+
+    /*
+        not sure about which structures to use yet
+    */
+    struct USER{
+        uint contributed; // number of times contributed or could be amount contributed
+        uint paid; // number of times bid Won
+        uint allowance; // how much they are allowed to withdraw, i.e if someone win , their allowance will go up by the bid.
+        bool alive; //needed to check if a member is indeed a member
+
+    }
+    mapping(address => USER) members; //using struct USER to keep track of contributions and paid, allowance and etc.
+    address[] memberAddress;    //this is the only way to iterate through all the member's address
+
+    mapping(address => bool) pendingJoinRequest; // this way , address can be used as index, if we use address[] , we'll have to go thru a whole array
+
+    //deposit and withdrawal related
+    uint totalAmountInContract;
+    //bidding related state variable
+    uint lowestBid;
+    address winnerAddress;
+
+
     modifier onlyForeman {
+        if(msg.sender != foreman)
+        throw;
+        _;
+    }
+    modifier beforeStart {
+        if(currentRound == 0)
+        throw;
         _;
     }
-    
+
     /**
      * Creates a new ROSCA in which every roundPeriod each person contributes
      * roundSum, for numRounds
      */
-    function ROSCA(
-        Period roundPeriod, 
-        uint roundSum, 
-        int16 numRounds, 
-        int8 minParticipants, 
-        int startTime, 
-        int16 feeInThousandths, 
-        address feeAddress);
-    
+    function rosca(
+        uint _roundPeriodInDays,
+        uint _roundSum, //i am assuming this is referring to number epochs to run
+        uint _numEpochs,
+        int8 _minParticipants,
+        uint _startTime,
+        int16 _feeInThousandths,
+        address _feeAddress)
+    {
+        roundPeriodInDays = _roundPeriodInDays;
+        if(roundSum < 1) throw; //if everyone is only required to contribute 0 , whats the point of rosca
+        roundSum = _roundSum;
+        if(_numEpochs < 1) throw;
+        numEpochs = _numEpochs;
+        if(_minParticipants < 2) throw;// there should be at least 2 people to make a group
+        minParticipants = _minParticipants;
+        if(_startTime < now) throw;
+        startTime = _startTime;
+        if(_feeInThousandths < 0) throw; //fee must be non-negative
+        feeToService = _feeInThousandths;
+        if(feeAddress != 0) //feeAddress shouldnt be empty(null)
+        feeAddress = _feeAddress;
+
+
+        //register foreman as a member of ROSCA and increment numRounds in a Epoch to 1
+        members[msg.sender] = USER({paid: 0 , contributed: 0, alive: true, allowance: 0});
+        memberAddress.push(msg.sender);
+        numRounds = 1;
+    }
+
+
+
+    function startRound()
```

Comment:
i believe we mentioned this during our meeting, at the end of each round, we need to call a function to get the winningBid and winningAddress otherwise there is no way for the contract to know when to end the round. Unless we use services like "Alarm clock"